Branch-SDK/Branch-SDK/Branch.h

```diff
@@ -399,7 +400,14 @@ typedef NS_ENUM(NSUInteger, BranchCreditHistoryOrder) {
 /// @name Deep Link Controller
 ///---------------------------
 
-- (void)registerDeepLinkController:(UIViewController <BranchDeepLinkingController> *)controller forKey:(NSString *)key;
+
+/**
+ Allow Branch to handle a view controller with options to push, present or show.
+ Note:
+ * If push option is used and the rootviewcontroller of window is not of type UINavigationViewController, than the sharing View controller would be presented automatically
+ * If UIViewController is pushed, than it wouldn't be popped on calling deepLinkingControllerCompleted delegate
+ */
+- (void)registerDeepLinkController:(UIViewController <BranchDeepLinkingController> *)controller forKey:(NSString *)key withOption:(BNCViewControllerOption)option;
```

Comment:
It's be important that you don't change the interface without warning. If a developer were to upgrade, their build would break. You need to make sure you continue to support the old method.